r-package/survlearners/R/surv_fl_grf.R

```diff
@@ -45,6 +45,7 @@ surv_fl_grf <- function(X, Y, W, D, t0, W.hat = NULL, cen.fit = "Kaplan-Meier",
     for (z in 1:k.folds) {
       c.fit <- survival::survfit(survival::Surv(Y[!fold.id == z], 1 - D[!fold.id == z]) ~ 1)
       C.hat[fold.id == z] <- summary(c.fit, times = U[fold.id == z])$surv
+      C.hat[fold.id == z][U[fold.id == z] < min(Y[!fold.id == z & D == 1])] <- 1
```

Comment:
Sorry, I think survival already is consistent, I don't remember what I was doing, but it must have been wrong because this looks as expected (!!!):

```
library(survival)
Y=c(1:5)
D=c(1, 1, 0, 1, 1)
km=survfit(Surv(Y,D) ~ 1)

C.hat.fold <- summary(km, times = Y)$surv
C.hat.fold
```

the only thing you'd have to take into account when using KM if you predict t0 at a large horizon somewhere with it:

```
> summary(km, times = 10)$surv
Error in findrow(fit, times, extend) : 
  no points selected for one or more curves, consider using the extend argument
```

Reply:
ok, glad it is clarified. Yeah, for the horizon beyond the available time points, KM cannot make a prediction